zfit_physics/models/pdf_novosibirsk.py

```diff
@@ -0,0 +1,169 @@
+
+    return znp.where(cond, result_gauss, result_novosibirsk)
+
+
+class Novosibirsk(zfit.pdf.BasePDF):
+    _N_OBS = 1
+
+    def __init__(
+        self,
+        mu,
+        sigma,
+        lambd,
+        obs,
+        *,
+        extended: Optional[ztyping.ExtendedInputType] = False,
+        norm: Optional[ztyping.NormInputType] = None,
+        name: str = "Novosibirsk",
+    ):
+        """Novosibirsk PDF.
+
+        The Novosibirsk function is a continuous probability density function (PDF) that is used to model
+        asymmetric peaks in high-energy physics. It is a theoretical Compton spectrum with a logarithmic Gaussian function.
+
+        .. math::
+            f(x;\\sigma, x_0, \\Lambda) = \\exp\\left[
+                -\\frac{1}{2} \\frac{\\left( \\ln q_y \\right)^2 }{\\Lambda^2} + \\Lambda^2 \\right] \\\\
+            q_y(x;\\sigma,x_0,\\Lambda) = 1 + \\frac{\\Lambda(x-x_0)}{\\sigma} \\times
+            \\frac{\\sinh \\left( \\Lambda \\sqrt{\\ln 4} \\right)}{\\Lambda \\sqrt{\\ln 4}}
+
+        Args:
+            mu: The peak of the distribution.
+            sigma: The width of the distribution.
+            lambd: The tail of the distribution.
+            obs: |@doc:pdf.init.obs| Observables of the
+               model. This will be used as the default space of the PDF and,
+               if not given explicitly, as the normalization range.
+
+               The default space is used for example in the sample method: if no
+               sampling limits are given, the default space is used.
+
+               The observables are not equal to the domain as it does not restrict or
+               truncate the model outside this range. |@docend:pdf.init.obs|
+            extended: |@doc:pdf.init.extended| The overall yield of the PDF.
+               If this is parameter-like, it will be used as the yield,
+               the expected number of events, and the PDF will be extended.
+               An extended PDF has additional functionality, such as the
+               ``ext_*`` methods and the ``counts`` (for binned PDFs). |@docend:pdf.init.extended|
+            norm: |@doc:pdf.init.norm| Normalization of the PDF.
+               By default, this is the same as the default space of the PDF. |@docend:pdf.init.norm|
+            name: |@doc:pdf.init.name| Human-readable name
+               or label of
+               the PDF for better identification.
+               Has no programmatical functional purpose as identification. |@docend:pdf.init.name|
+        """
+        params = {"mu": mu, "sigma": sigma, "lambd": lambd}
+        super().__init__(obs=obs, params=params, name=name, extended=extended, norm=norm)
+
+    def _unnormalized_pdf(self, x):
+        mu = self.params["mu"]
+        sigma = self.params["sigma"]
+        lambd = self.params["lambd"]
+        return novosibirsk_pdf(x, peak=mu, width=sigma, tail=lambd)
```

Comment:
Just for the naming, I like the greek letters, but shouln'd we have them maybe consistently, also in the `novosibirsk_pdf`? It doesn't have to be, I'll easily buy the argument that we keep the same names as used i.e. in the roofit implementation - on the other hand, if we want to make that function more public, which I think is a good idea in the long run, it seems most likely confusing to a user that the PDF object and the function would take different arguments, wouldn't it?

Reply:
Yes, the idea was to have the same variables as root's c++ code internally because the implementation really follows that. I would like to keep it like that unless you strongly opose.

Reply:
I think the question is rather, how to expose the function to the user? If we expose it, wouldn't this be confusing to have _different_ parameters in the function than in the pdf?
So the internals, makes sense. But for the user? (we don't expose it really currently, but users will still pick it up most likely.
What about just adding comments that translate?
So overall, not a very strong opinion, but I think we just should not mix the _internals_ vs _externals_, and if we assume that the function API should be, in general, available to the user, or for  future, it would make sense to have this consistent?
It's not exposed yet, oc, but if we would expose it at least, we would need to wrap it, right?

Reply:
I suggest we rename or wrap in the future if that happens

Reply:
What about: we name the parameters the same as in the pdf and _inside_ the function we do a `mean=mu` etc, with the comment to "follow the RooFit implementation"? With tha we don't need to ever rename and we keep the implementation the same?

(renaming params later on is just a bad thing in general: if somebody used the code already, it will break the code)

Reply:
That would satisify all needs I guess?